File: backend/src/main/java/es/udc/pashop/backend/model/services/CatalogService.java
package es.udc.pashop.backend.model.services;

import java.util.List;

import es.udc.pashop.backend.model.exceptions.InstanceNotFoundException;
import es.udc.pashop.backend.model.entities.Category;
import es.udc.pashop.backend.model.entities.Product;

public interface CatalogService {
	
	List<Category> findAllCategories();
	
	Product findProductById(Long id) throws InstanceNotFoundException;
	
	Block<Product> findProducts(Long categoryId, String keywords, int page, int size);

}

File: frontend/src/backend/catalogService.js
import {appFetch} from './appFetch';

export const findAllCategories = async () => await appFetch('GET', '/catalog/categories');

export const findProducts = async ({categoryId, keywords, page}) => {

    let path = `/catalog/products?page=${page}`;

    path += categoryId ? `&categoryId=${categoryId}` : "";
    path += keywords.length > 0 ? `&keywords=${encodeURIComponent(keywords)}` : "";

    return await appFetch('GET', path);

}

export const findProductById = async id => await appFetch('GET', `/catalog/products/${id}`);
